### PR TITLE
Documentation: Update link-no-children error page for new link

### PR DESCRIPTION
**Pull request by divslinger**
The example still has an `<a>` inside the `<Link>` which was deprecated with next 13.

Current page: https://nextjs.org/docs/messages/link-no-children

## Documentation / Examples

- [x] Make sure the linting passes by running `pnpm build && pnpm lint`
- [x] The "examples guidelines" are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md)

### CODE REVIEW

File: errors/link-no-children.md [hunk 1]
```diff
@@ -27,7 +27,9 @@ import Link from 'next/link'
 
 export default function Home() {
   return (
-    <Link href="/about">
+    <Link href="/about">To About</Link>
+    // or
+    <Link href="/about" legacyBehavior>
       <a>To About</a>
     </Link>
```
Comment (ijjk) [inline errors/link-no-children.md:34]:
```suggestion
    <>
      <Link href="/about">To About</Link>
      // or
      <Link href="/about" legacyBehavior>
        <a>To About</a>
      </Link>
    </>
```

**Review by ijjk — APPROVED** [review]

**Review by ijjk — CHANGES_REQUESTED** [review]
It looks like this change isn't necessary as this document is specific to the legacy behavior and the `React.children.only` restriction isn't present in the new handling. Thanks for the PR though. 

x-ref: https://github.com/vercel/next.js/blob/aa614635d6520cc9dac3cc91f145446d1317611f/packages/next/src/client/link.tsx#L465-L479

**Comment by divslinger** [comment]
I'm sorry @ijjk I can't quite follow.
The error message and link is only shown to the user when `legacyBehavior` is true and there are no children.

However, the current doc doesn't have the `legacyBehavior` attribute.

The only viable fixes are the one's from my PR: Either keep the `legacyBehavior` attribute and use an `<a>` child, or remove `legacyBehavior` and add some other child.

To recap, all things wrong with the current version:

1. The code under “[Why This Error Occurred](https://nextjs.org/docs/messages/link-no-children#why-this-error-occurred)” doesn't have the `legacyBheavior` attribute, but this link will only be logged to the user when it is present.
2. The solution is currently missing the `legacyBehavior` tag for it to be used with an `<a>` child
3. The solution is missing an example without `legacyBehavior` tag

2 and 3 are fixed by my PR. 
[1 is also already fixed, but on my fork](https://github.com/vercel/next.js/commit/3bd16da6c808ddb66d09eceb730cafabf1604673), I can't update the PR if it's closed. 
If you agree with my assessment, please reopen, so I can update the PR.

**Comment by ijjk** [comment]
Hmm if we specify the version `legacyBehavior` was added/defaulted this change seems fine

**Review by timneutkens — DISMISSED** [review]

**Comment by timneutkens** [comment]
Thanks!

**Pull request merged**